installer/pkg/components/agent-smith/daemonset.go

```diff
@@ -18,6 +18,12 @@ import (
 func daemonset(ctx *common.RenderContext) ([]runtime.Object, error) {
 	labels := common.DefaultLabels(Component)
 
+	// todo(sje): regenerate this with actual configmap checksum
+	checksum, err := common.RandomString(20)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Actually producing the checksum is about as few lines of code as the random string. After seeing the `RandomString` approach, I've taken the liberty and added a commit in https://github.com/gitpod-io/gitpod/pull/6550 which implements actual checksums.

Reply:
Ok. I'll remove this commit then and defer to yours